### PR TITLE
Fix display of docker errors during storage build

### PR DESCRIPTION
**Pull request by madkinsz**
## Summary

Fixes missing display of docker error messages during image build. On docker-py version 4.3.1 and Docker version 19.03.13, build 4484c46d9d the message for a syntax error in the dockerfile was in a format we did not check for. This gave me a `ValueError` suggesting a healthcheck had failed when really there was a dockerfile syntax error that was hidden.

## Changes

- Adds a check for "message" in the docker build output parsing

## Importance

Helps debug bad Docker storage setups

## Checklist

This PR:

- ~[ ] adds new tests (if appropriate)~
- [x] adds a change file in the `changes/` directory (if appropriate)
- ~[ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)~

### CODE REVIEW

**Comment by joshmeek** [comment]
Lgtm @madkinsz do you want to add a `/changes` entry for this? (with `[ci skip]` in the commit message in order to avoid having to rerun tests)

**Comment by madkinsz** [comment]
> Lgtm @madkinsz do you want to add a `/changes` entry for this? (with `[ci skip]` in the commit message in order to avoid having to rerun tests)

Yeah I'll add a changes entry but I was actually thinking of adding test coverage as well, should I not bother?

**Review by joshmeek — APPROVED** [review]

**Review by jcrist — APPROVED** [review]

File: src/prefect/environments/storage/docker.py [hunk 1]
```diff
@@ -653,9 +653,14 @@ def _parse_generator_output(self, generator: Iterable) -> None:
...
-                        print(output.strip("\n"))
+                if not line:
+                    continue
+                parsed = json.loads(line)
+                # Parse several possible schemas
+                output = (
+                    parsed.get("stream")
+                    or parsed.get("message")
+                    or parsed.get("errorDetail", {}).get("message")
+                ).strip("\n")
```
Comment (mgsnuno) [inline src/prefect/environments/storage/docker.py:664]: if output is `None`, `strip` fails.

**Review by mgsnuno — COMMENTED** [review]
@joshmeek with latest prefect 0.13.17 I get now the error `AttributeError: 'NoneType' object has no attribute 'strip'` while buidling Docker storage.

**Comment by madkinsz** [comment]
> @joshmeek with latest prefect 0.13.17 I get now the error `AttributeError: 'NoneType' object has no attribute 'strip'` while buidling Docker storage.

Thanks for the report, I'll submit a fix today.

**Pull request merged**